### DISCUSSION TITLE
New releases are not detected for Maven dependencies hosted in Github's private registry

### DISCUSSION THREAD

**Discussion by marcospassos** [renovatebot/renovate#13023] [Request Help]
We've dozens of Maven repositories hosted in a private Github registry that suddenly stopped detecting new releases. I've analyzed the log in detail, and it seems like Renovate can't find the latest three releases:

![image](https://user-images.githubusercontent.com/943036/145313865-371b7c4d-fdc9-4487-ae87-7a88fec0a3ba.png)

Although the package is at v0.5.2, renovate only find an outdated version 3 releases behind:

![image](https://user-images.githubusercontent.com/943036/145314398-76ab768c-aa25-4408-8f94-aa109492599b.png)

![image](https://user-images.githubusercontent.com/943036/145314439-64519a51-e00c-45ce-bafe-afa275211f4f.png)

This projects, in fact, have 16 releases:

![image](https://user-images.githubusercontent.com/943036/145313952-3e0542ef-41c3-41e7-903f-c2a76bde5902.png)

That's the config we're using:

```js
{
  "extends": [
    "config:base"
  ],
  "labels": ["maintenance"],
  "reviewers": ["croct-tech/java"],
  "masterIssue": true,
  "automergeStrategy": "squash",
  "automergeType": "pr",
  "packageRules": [
    {
      "matchManagers": ["maven"],
      "matchPackagePrefixes": ["com.croct"],
      "registryUrls": [
        "https://maven.pkg.github.com/croct-tech/REPOSITORY/"
      ],
      "schedule": ["at any time"]
    },
    {
      "matchUpdateTypes": ["minor", "patch"],
      "matchCurrentVersion": "!/^0/",
      "automerge": true
    }
  ],
  "schedule": ["before 5am every weekday"],
  "prHourlyLimit": 0,
  "hostRules": [
    {
      "hostType": "maven",
      "matchHost": "https://maven.pkg.github.com/",
      "username": "<redacted>",
      "encrypted": {
        "token": "<redacted>"
      }
    }
  ]
}
```

Besides Maven, we also use Renovate for PHP and Node. Everything is working as usual. For Maven, however, it's not working anymore.

**Comment by rarkins** [comment]
The maven datasource should merge results between registries. If you're seeing partial results, would that be because only some releases are published on the alternative registries? Or is it certain that it's returning incomplete results from your private registry?

**Reply by marcospassos** [reply]
That's the latter.

**Reply by rarkins** [reply]
It's strange to find only 13 of 16 releases. e.g. unlikely to be a pagination problem when the total is so small. Can you use `curl` or similar to manually query the same URL with the same credentials and post the results here?

**Reply by marcospassos** [reply]
That's what I tried. Sending requests straight to the GH API returns all 16 versions. 

**Update**

Does the version lookup have some sort of caching mechanism? About 10 hours later, Renovate detected the new versions and opened the PR (notice that it correctly recognize the configuration to open PR's anytime).

![image](https://user-images.githubusercontent.com/943036/145399790-70b68eac-2f0d-4b34-8af4-f8b6970d189f.png)

As the log showed, when I started this discussion (~30 minutes after releasing new versions), Renovate couldn't detect the latest versions, but only several hours later, which makes me wonder if there is any caching layer between that could cause such delay.

**Reply by rarkins** [reply]
Yes, there is caching. I didn't realize you were commenting so close to the releases being made available. With a 1 hour cache time and Renovate running every 1-3 hours, it could explain everything

**Reply by marcospassos** [reply]
Is there any way to narrow this cache window using the hosted or self-hosted version? It's has a massive impact on our productivity since we rely on Renovate to cascade the updates downstream.

**Reply by marcospassos** [reply]
To be more precise: the bot only opened the PR about 10h later. That's a lot!

**Reply by viceice** [reply]
how are you running renovate? are you preserving renovate `baseDir` between runs?

**Reply by marcospassos** [reply]
> Yes, there is caching. I didn't realize you were commenting so close to the releases being made available. With a 1 hour cache time and Renovate running every 1-3 hours, it could explain everything

Not exactly. I forced the bot to check for updates using the Dashboard Issue. The problem is that the lookup is cached, so the result doesn't include the new releases.

**Reply by marcospassos** [reply]
@viceice, we're using the managed version, not the self-hosted. But if using the self-hosting allows disabling this cache, we'll probably move to the self-hosted solution.

**Reply by rarkins** [reply]
I think this is the caching logic for maven: https://github.com/renovatebot/renovate/blob/ec6077d343c5a3ac054c4821d01a8fae058ba1ff/lib/datasource/maven/index.ts#L85-L87

So I'm not sure why you'd be affected by caching at all

**Reply by marcospassos** [reply]
Yes, something really weird here. And as I said, this is something recent. About a month ago, the releases were detected in near real-time.
